### ISSUE TITLE
v6.16.2 fails to mine Randomx with RX580

### ISSUE THREAD

**Issue by UselessGuru** [xmrig/xmrig#2803]
![image](https://user-images.githubusercontent.com/30080938/145178789-5dbba894-d1ef-46d2-a9d0-79ef61c11e9a.png)

RX5700 with same driver works fine

**Comment by Titaniumtown** [comment]
why are you trying to mine monero on your gpus lol

**Comment by Spudz76** [comment]
Probably mixed Ellesmere and Navi needs driver no newer than 21.6.1 or 21.7.2

**Comment by dastardlyman12** [comment]
uselessguru we respect you. what im doing is mining eth on my rx580 and converting it into the durrty

**Comment by Lonnegan** [comment]
ETH or ERGO or XHV. But mining RandomX via GPU is a waste of energy or profits given away.

**Comment by UselessGuru** [comment]
> why are you trying to mine monero on your gpus lol

I am the co-developer of www.nemosminer.com. Our aim is to support all POSSIBLE algos. The program will chose the best algorithm (and this will never be Monero on GPU's). But If don't include it then there will be questions like 'Why don't you support xyz...'.

**Comment by SChernykh** [comment]
Try to set `"gcn_asm":false,` in config.json for that GPU. Maybe new AMD drivers and Windows 11 are not compatible with ASM implementations.

**Comment by UselessGuru** [comment]
@SChernykh

Is there also a command line parameter for "gcn_asm":false, ?

**Comment by Titaniumtown** [comment]
> > why are you trying to mine monero on your gpus lol
> 
> I am the co-developer of [www.nemosminer.com](http://www.nemosminer.com). Our aim is to support all POSSIBLE algos. The program will choose the best algorithm (and this will never be Monero on GPU's). But If don't include it then there will be questions like 'Why don't you support xyz...'.

MoneroOcean already exists lol....

**Comment by Unnamed-Stranger** [comment]
> @SChernykh
> 
> Is there also a command line parameter for "gcn_asm":false, ?

Yes, it can be specified in the config file, in the "profile" object.
Example:

`"opencl":
{
"enabled":true,
"loader":"Some_Path_Here",
"adl":true,
"cache":false,
"platform":"AMD",
"rx":
[{
"intensity":2097152,
"worksize":160,
"gcn_asm":true,
"index":0},
{
"intensity":2097152,
"worksize":160,
"gcn_asm":true,
"index":1}
]}`

**Comment by Spudz76** [comment]
> Yes, it can be specified in the config file

That's not a command line parameter then, is it?
